File: 2155.cpp
struct Solution {
    inline vector<int> maxScoreIndices(vector<int>& nums) {
        vector<int> res(1, 0);
        int score = 0, high = 0;
        for (int i = 0; i < nums.size(); ++i){
            if (nums[i] == 0){
                ++score;
            } else {
                --score;
            }
            if (score > high){
                high = score;
                res.clear();
            }
            if (score == high){
                res.push_back(i + 1);
            }
        }
        return res;
    }
};

File: sol/2155.py
class Solution:
    def maxScoreIndices(self, nums: List[int]) -> List[int]:
        res = [0]
        score = 0
        high = score
        for i, x in enumerate(nums):
            if x == 0:
                score += 1
            else:
                score -= 1
            if score > high:
                high = score
                res.clear()
            if score == high:
                res.append(i + 1)
        return res
